### DISCUSSION TITLE
Different outer and inner gaps.

### DISCUSSION THREAD

**Discussion by lominoss-git** [qtile/qtile#2798] [Q&A]
I'm just asking if there is any way to have different inner and outer gap sizes. Like indicated in this screenshot:
![VirtualBox_Pop!_OS 64-bit - Daily Driver_16_09_2021_12_10_52](https://user-images.githubusercontent.com/79030093/133608060-9b2a5420-0112-4b7c-9ded-ec81bd466d61.png)

**Comment by m-col** [comment]
The gaps on the outside you can specify using `bar.Gap`, which you pass to the `Screen`(s) in your config, whereas the gaps around the windows are not part of the `Screen` but rather the `Layout`. For example, if you wanted inner margins on the `Columns` layout:
```py
layouts = [Columns(margin=10)]
screens = [Screens(top=bar.Gap(10))]
```

**Reply by lominoss-git** [reply]
I don't know how to implement your solution in my existing config:
```python
screens = [
    Screen(
        top=bar.Bar(
            [
                widget.Spacer(),
                # Spotify widget:
                widget.Mpris2(
                    name = 'spotify',
                    objname = "org.mpris.MediaPlayer2.spotify",
                    display_metadata = ['xesam:artist', 'xesam:title'],
                    scroll_chars = None,
                    stop_pause_text = None,
                    mouse_callbacks = {
                        "Button1": lambda: qtile.cmd_spawn("spotifycli --playpause"),
                        "Button4": lambda: qtile.cmd_spawn("spotifycli --next"),
                        "Button5": lambda: qtile.cmd_spawn("spotifycli --prev"),
                    }
                ),
                widget.TextBox(text="&lt;", foreground="#9E9E9E", padding=18),
                widget.Volume(fmt="Vol: {}"),
                widget.TextBox(text="&lt;", foreground="#9E9E9E", padding=18),
                widget.Clock(format='%A, %B %d'),
                widget.TextBox(text="&lt;", foreground="#9E9E9E", padding=18),
                widget.Clock(format='%I:%M:%S %p '),
            ],
            50, background = "#FFFFFF",
        ),
    ),
]
```

**Reply by elParaguayo** [reply]
To add gaps, you'd do something like this:
``` python
screens = [
    Screen(
        top=bar.Bar(
            [
                widget.Spacer(),
                # Spotify widget:
                widget.Mpris2(
                    name = 'spotify',
                    objname = "org.mpris.MediaPlayer2.spotify",
                    display_metadata = ['xesam:artist', 'xesam:title'],
                    scroll_chars = None,
                    stop_pause_text = None,
                    mouse_callbacks = {
                        "Button1": lambda: qtile.cmd_spawn("spotifycli --playpause"),
                        "Button4": lambda: qtile.cmd_spawn("spotifycli --next"),
                        "Button5": lambda: qtile.cmd_spawn("spotifycli --prev"),
                    }
                ),
                widget.TextBox(text="&lt;", foreground="#9E9E9E", padding=18),
                widget.Volume(fmt="Vol: {}"),
                widget.TextBox(text="&lt;", foreground="#9E9E9E", padding=18),
                widget.Clock(format='%A, %B %d'),
                widget.TextBox(text="&lt;", foreground="#9E9E9E", padding=18),
                widget.Clock(format='%I:%M:%S %p '),
            ],
            50, background = "#FFFFFF",
        ),
        right=bar.Gap(10),
        left=bar.Gap(10),
        bottom=bar.Gap(10)
    ),
]
```

**Reply by elParaguayo** [reply]
Also look at #2843 if you want to change these with different layouts.

**Reply by hamensman** [reply]
How about in the ArcoLinux style configuration (which I pretty much use)? It's organised quite differently from the standard.
https://github.com/arcolinux/arcolinux-qtile/blob/master/etc/skel/.config/qtile/config.py

**Reply by elParaguayo** [reply]
Which bit are you referring to? The layouts in that config use the `margin` setting which is what m-col referred to.

To be honest, I find the ArcoLinux config a bit confusing, there's a lot of unnecessary wrapping of things in functions. `widgets_list` is defined in `init_widgets_list`. There are then two separate functions `init_widgets_screen1` and `init_widgets_screen2` that just return the output of `init_widgets_list` which is `widgets_list` 😕 It gets even better as there are two variables called `widgets_screen1` and `widgets_screen2` which take the return value of `init_widgets_screen1` and `init_widgets_screen2` respectively but these variables are never used.

I wonder if they've done this because they don't want some widgets to be mirrored but it feels overly complicated to me. If there is a particular issue they're trying to structure around then I'd be interested in hearing it and seeing if we can help implement a fix that would allow them to simplify their config file.

**Reply by hamensman** [reply]
> Which bit are you referring to?

Apologies. I mean the:
```
right=bar.Gap(10),
left=bar.Gap(10),
bottom=bar.Gap(10)
```
part.
> To be honest, I find the ArcoLinux config a bit confusing, there's a lot of unnecessary wrapping of things in functions.

Yeah. I'll probably just change my config to more of the standard orientation. The ArcoLinux style has made trying to set up the above rather difficult. Overall it's just very cumbersome to keep track of all these functions...

> ...but it feels overly complicated to me

And don't even get me started on ArcoLinux's website... (saying this as someone who used ArcoLinux for at least 4 months)
